src/openvic-simulation/economy/production/ArtisanalProducer.cpp

```diff
@@ -1,13 +1,116 @@
 #include "ArtisanalProducer.hpp"
 
+#include "openvic-simulation/economy/GoodDefinition.hpp"
+#include "openvic-simulation/economy/trading/BuyResult.hpp"
+#include "openvic-simulation/economy/trading/SellResult.hpp"
+#include "types/fixed_point/FixedPoint.hpp"
+
 using namespace OpenVic;
 
 ArtisanalProducer::ArtisanalProducer(
-	ProductionType const& new_production_type,
+	MarketInstance& new_market_instance,
+	ModifierEffectCache const& new_modifier_effect_cache,
+	Pop& new_pop,
 	GoodDefinition::good_definition_map_t&& new_stockpile,
-	fixed_point_t new_current_production,
-	GoodDefinition::good_definition_map_t&& new_current_needs
-) : production_type { new_production_type },
-	stockpile { std::move(new_stockpile) },
-	current_production { new_current_production },
-	current_needs { std::move(new_current_needs) } {}
+	ProductionType const& new_production_type,
+	fixed_point_t new_current_production
+) : market_instance { new_market_instance },
+	modifier_effect_cache { new_modifier_effect_cache },
+	pop { new_pop },
+	stockpile { new_stockpile },
+	production_type { new_production_type },
+	current_production { new_current_production }
+	{}
+
+void ArtisanalProducer::artisan_tick() {
+	GoodDefinition::good_definition_map_t goods_to_buy_and_max_price { };
```

Comment:
Should we merge artisanal inputs and pop needs?
Can pops consume their own produce?
How is money allocated between inputs and needs?